### PR TITLE
DAOS-1871-4 Object: Add csum error logging.

### PR DESCRIPTION
**Pull request by jgmoore-or**
Add checksum error logging for key validation and for checksum validation on update.

Signed-off-by: Joseph Moore <joseph.moore@intel.com>

### CODE REVIEW

File: src/object/srv_obj.c [hunk 1]
```diff
@@ -876,6 +876,22 @@ obj_singv_ec_rw_filter(struct obj_rw_in *orw, daos_iod_t *iods, uint64_t *offs,
 	}
 }
 
+/* Call internal method to increment CSUM media error. */
+static void
+obj_log_csum_err(void)
+{
+	struct dss_module_info	*info = dss_get_module_info();
+	struct bio_xs_context	*bxc  = info->dmi_nvme_ctxt;
+
```
Comment (jolivier23) [inline src/object/srv_obj.c]: I've been seeing a lot of SDL patches lately checking NULL conditions.  do we need a D_ASSERT(info != NULL) here?

Comment (jgmoore-or) [reply src/object/srv_obj.c]: I can add the assert.  The dss_module_info is initialized during module load.  I don't think we'd get this far in the server if it failed.  However, an assert failure is preferable to a seg fault.

Comment (jolivier23) [reply src/object/srv_obj.c]: It's more for the security scans than whether it can happen or not

**Comment by rpadma2** [comment]
mdtestsmall test failed similar to PR1943. You may want to upload this test results to bug DAOS-4294. I think the problem is with dfuse(or daos) implementation. I suspect it is not specific to checksum enablement.

**Pull request merged**